controllers/rest/middleware_context.go

```diff
@@ -34,6 +35,87 @@ func Context(api *RESTAPI, projection projections.Projection, commandDispatcher
 			if accountID != "" {
 				cc = context.WithValue(cc, weosContext.ACCOUNT_ID, accountID)
 			}
+			//use x-session to get the properties needed for the context
+			if tsessionParams, ok := operation.ExtensionProps.Extensions[SessionExtension]; ok {
+				sessionvalues := map[string]interface{}{}
+				var sessionParams map[string]interface{}
+				err = json.Unmarshal(tsessionParams.(json.RawMessage), &sessionParams)
+				if err != nil {
+					api.EchoInstance().Logger.Errorf("unexpected error unmarshalling x-session")
+					return NewControllerError("unexpected error unmarshalling x-session", err, http.StatusBadRequest)
+				}
+				//get cookies from the request
+				cookies := c.Cookies()
+				if len(cookies) == 0 {
+					api.EchoInstance().Logger.Errorf("unexpected error no cookies were found")
+					return NewControllerError("unexpected error no cookies were found", nil, http.StatusBadRequest)
+				}
+				//get session from api
+				sessionStore := api.GetSessionStore()
+				if sessionStore == nil {
+					api.EchoInstance().Logger.Errorf("unexpected error no session store was found on api")
+					return NewControllerError("unexpected error no session store was found on api", nil, http.StatusInternalServerError)
+				}
+				var session *sessions.Session
+				for _, cookie := range cookies {
+					//get session by name
+					sess, err := sessionStore.Get(c.Request(), cookie.Name)
+					if err != nil {
+						api.EchoInstance().Logger.Errorf("unexpected error %s", err)
+						return NewControllerError(fmt.Sprintf("unexpected error %s", err), err, http.StatusBadRequest)
+					}
+					//check if session id is the same as the id in the cookie
+					if sess != nil && sess.ID == cookie.Value {
+						session = sess
+						break
+					}
+				}
+				if session == nil {
+					api.EchoInstance().Logger.Errorf("unexpected error no session found with cookie name")
+					return NewControllerError("unexpected error no session found with cookie name", nil, http.StatusBadRequest)
+				}
+				for key, value := range sessionParams["properties"].(map[string]interface{}) {
```

Comment:
there is already a function ConvertStringToType for doing this, why not use that?